src/toil/provisioners/clusterScaler.py

```diff
@@ -331,6 +347,26 @@ def tryRun(self):
                                 estimatedNodes)
                     totalNodes = self.scaler.provisioner.setNodeCount(numNodes=estimatedNodes,
                                                                       preemptable=self.preemptable)
+                    
+                    # if we were scaling up the number of preemptable nodes and failed to
+                    # meet our target, we need to update the slack so that non-preemptable
+                    # nodes will be allocated and we won't block. if we _did_ meet our target,
+                    # we need to reset the slack to 0
+                    if totalNodes < estimatedNodes and self.preemptable:
+                        
+                        # slack is derived from the delta (the number of nodes we did _not_ allocate)
+                        # times a preference for preemptable nodes
+                        require(1.0 >= self.scaler.config.slackPreemptablePreference >= 0.0,
+                                "Slack preference for preemptable nodes (%f) must be >= 0.0 and <= 1.0",
+                                self.scaler.config.slackPreemptablePreference)
+                        nonPreemptablePreference = 1.0 - self.scaler.config.slackPreemptablePreference
+                        delta = estimatedNodes - totalNodes
+                        _preemptableQueueSlack = int(round(delta * nonPreemptablePreference))
```

Comment:
Yeah, that sounds reasonable. Shouldn't be a hard change. I'll do that tonight/tomorrow.